File: WEB/FootageAngular/src/app/model/interfaces/producto.ts
export interface ProductoResponse {
    id:          number;
    titulo:      string;
    descripcion: string;
    precio:      number;
    intercambio: boolean;
    original:    boolean;
    foto:       string;
    usuario:     Usuario;
    seccion:     Categoria;
    categoria:   Categoria;
    marca:       Categoria;
    talla:       Talla;
    meGustas:    Usuario[];
}

export interface Categoria {
    id:     number;
    nombre: string;
    imagen: string;
}

export interface Usuario {
    id:           number;
    nombre:       string;
    apellidos:    string;
    username:     string;
    email:        string;
    avatar:       string;
    localizacion: string;
    premium:      boolean;
}

export interface Talla {
    id:        number;
    nombre:    string;
}

File: API/FootageBackend/src/main/java/com/danieloliva/FootageBackend/model/Producto.java
package com.danieloliva.FootageBackend.model;

import com.danieloliva.FootageBackend.usuario.model.Usuario;
import lombok.*;
import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "productos")
public class Producto implements Serializable {

    @Id
    @GeneratedValue
    private Long id;

    private String titulo;

    @Lob
    private String descripcion;

    private double precio;

    private boolean intercambio;

    private boolean original;

    private String foto;

    @ManyToOne
    @JoinColumn(name = "usuario_id", updatable = false, nullable = false)
    private Usuario usuario;

    @ManyToOne
    @JoinColumn(name = "seccion_id")
    private Seccion seccion;

    @ManyToOne
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;

    @ManyToOne
    @JoinColumn(name = "marca_id")
    private Marca marca;

    @ManyToOne
    @JoinColumn(name = "talla_id")
    private Talla talla;

    @Builder.Default
    @OneToMany(mappedBy = "producto", cascade = {CascadeType.REMOVE})
    private List<MeGusta> meGustas = new ArrayList<>();

}
